src/qrscanner.py

```diff
@@ -168,9 +170,33 @@ def update(self):
             if len(res) > 0:
                 self.data += res.decode('utf-8')
             if self.is_done() and self.callback is not None:
+                err_msg = None
                 data = self.data[:-len(self.EOL)]
                 self.reset()
-                self.callback(data)
+                # is this animated QR code
+                if data.startswith("part") and self.callback_animated_qr is not None:
```

Comment:
here we should probably check if we have 2 words first and then check if the first word starts with "part"
  * this way we certainly avoid collisions so we could also use the shorter version: "p1of3  payload"

Reply:
Makes sense. 
We could use
```py
import re

m = re.match("part\d+of\d+ (.*)",data)
# m.group(0) would be full text if it matches
# m.group(1) will be our payload
# m is None if it didn't have "partMofN"
if m and self.callback_animated_qr:
# ...
```

Reply:
I combined stuff from both comments into the code. Just doing regex on a medium sized QR code results in "maximum recursion depth exceeded"

Reply:
ok, it's another annoying constrain of micropython... Now I remember that I also had this issue some time ago.
Screw regexps, let's just use `data.split(" ")`, check that the first part starts with `part` and split with `of` and get the numbers.

Reply:
fixed